lib/galaxy/jobs/__init__.py

```diff
@@ -1929,7 +1929,7 @@ def fail(message=job.info, exception=None):
         # Finally set the job state.  This should only happen *after* all
         # dataset creation, and will allow us to eliminate force_history_refresh.
         job.set_final_state(final_job_state, supports_skip_locked=self.app.application_stack.supports_skip_locked())
-        if not job.tasks:
+        if not job.tasks and job_metrics_directory:
```

Comment:
I don't think this is right ? As dannon said, you only need to pass this if the root job working directory is not the one to use (that's the case for pulsar), this would break job metric collection for non pulsar.

Reply:
```suggestion
        if not job.tasks and (job_metrics_directory or job.job_runner_name != "PulsarJobRunner"):
```

?